### DISCUSSION TITLE
[Help]: Is it possible to change slide widths + scale when swiping?

### DISCUSSION THREAD

**Discussion by keithengg** [davidjerleke/embla-carousel#978] [Help]
### Summary

<img width="879" alt="Screenshot 2024-08-17 at 5 25 34 PM" src="https://github.com/user-attachments/assets/0dce11a2-1088-452d-b575-2797d8f664fd">

I am trying to achieve something like this where 2 slides off to the side takes up equal widths while the center "active" slide is bigger.

I followed the "Scale" example which works great but sadly trying to apply the same technique for changing slide size causes the Carousel to break. Anyone has some suggestions on workarounds? 

### If applicable, which variants of Embla Carousel are relevant to this question?

- [ ] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

**Comment by Joobs** [comment]
@keithengg did you ever find out a solution?

I've never done this effect, but I have a hacky suggestion. I've once used an `opacity: 0` carousel in the background that would capture drag information. Then in the foreground had markup that controlled a non standard looking carousel (it was a wheel of content), and passed position data between the two, but it was a lot of effort as it required hooking into the internal engine.

**Comment by tayteboss** [comment]
@davidjerleke. I am also stuck on this issue.

@keithengg Did you find a solution? 

When scrolling I want the center image to grow in width smoothly. Changing the flex: 0 0 value; breaks the carousel

**Comment by davidjerleke** [comment]
@keithengg, @Joobs, @tayteboss — adjusting slide sizes while the carousel is animating will reset the carousel. This kind of change isn't supported while the carousel is animating, as any ongoing animation will be interrupted and the carousel will reset. A workaround could be to use `transform: scale` instead. It may or may not work for your use case.

However, you can **disable the internal resize logic** (which triggers the reset) and implement your own instead. That way, you have full control over when the resize should be triggered—allowing you to avoid it when slide sizes change during transitions:

```tsx
const [emblaRef, emblaApi] = useEmblaCarousel({
  watchResize: false,
})
```

Just keep in mind that the internal resize logic calls `emblaApi.reInit()` when sizes change, which is good to know if you're writing your own resize handling.
